astropy/utils/tests/test_decorators.py

```diff
@@ -801,51 +815,58 @@ def docstring2():
     def testfunc3():
         """= 4 / 2 * test"""
 
-    assert inspect.getdoc(testfunc3) == "test + test = 4 / 2 * test"
+    expected_doc = "test + test = 4 / 2 * test" if sys.flags.optimize < 2 else None
+
+    assert inspect.getdoc(testfunc3) == expected_doc
 
 
 def test_format_doc_selfInput_simple():
     # Simple tests with self input
 
-    # Self input while the function has no docstring raises an error
-    with pytest.raises(ValueError):
+    if sys.flags.optimize < 2:
+        # Self input while the function has no docstring raises an error
+        with pytest.raises(ValueError):
 
-        @format_doc(None)
-        def testfunc_fail():
-            pass
+            @format_doc(None)
+            def testfunc_fail():
+                pass
 
     # Test that it keeps an existing docstring
     @format_doc(None)
     def testfunc_1():
         """not test"""
 
-    assert inspect.getdoc(testfunc_1) == "not test"
+    expected_doc = "not test" if sys.flags.optimize < 2 else None
+    assert inspect.getdoc(testfunc_1) == expected_doc
 
 
 def test_format_doc_selfInput_format():
     # Tests with string input which is '__doc__' (special case) and formatting
 
-    # Raises an indexerror if not given the formatted args and kwargs
-    with pytest.raises(IndexError):
+    if sys.flags.optimize < 2:
```

Comment:
Same comment as https://github.com/astropy/astropy/pull/17549/files#r1889354376 (sorry for the spam)